python/teams_ai/moderator/moderator.py

```diff
@@ -0,0 +1,38 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from abc import ABC, abstractclassmethod
+from typing import Optional
+
+class Moderator(ABC):
+    """
+    interface implemented by all moderators
+    """
+
+    @abstractclassmethod
+    async def review_prompt(input: str) -> Optional[Plan]:
```

Comment:
Is there a reason we name this method `review_prompt` when we're actually reviewing the user input. Unless, is it the rendered prompt we are rendering?

Reply:
Adding comment for tracking :)

Reply:
Do you have any ideas about the naming?
I didn't sync this with Steve. But since what users sent to OpenAI is called `prompt`, maybe we can keep using this name? The proposal #203 also keeps using `reviewPrompt` as function name for js SDK.